docs/types/literal-suffixes.rst

```diff
@@ -0,0 +1,243 @@
+|                                                             | - :ref:`String literals<string_literals>`                      |
+|                                                             | - :ref:`Unicode literals<unicode_literals>`                    |
++-------------------------------------------------------------+----------------------------------------------------------------+
+| ``string``                                                  | - :ref:`String literals<string_literals>`                      |
+|                                                             | - :ref:`Unicode literals<unicode_literals>`                    |
++-------------------------------------------------------------+----------------------------------------------------------------+
+
+For two-parameter suffix functions, the first parameter (representing the mantissa) can be of any integer type.
+The second parameter (the exponent) must be of an unsigned integer type.
+
+.. note::
+    40-digit literals prefixed with ``0x`` such as, for example, ``0xdCad3a6d3569DF655070DEd06cb7A1b2Ccd1D3AF``
+    always represent ``address`` literals in the language.
+    To invoke a suffix accepting ``bytes20`` you must use one of the other literal kinds implicitly
+    convertible to ``bytes20``, e.g., a hexadecimal string literal
+    (``hex"dCad3a6d3569DF655070DEd06cb7A1b2Ccd1D3AF"``).
+
+.. note::
+    Suffix functions accepting ``address payable`` are not allowed since address literals are never payable.
+
+Suffix functions may not accept or return reference types with ``storage`` or ``calldata`` locations.
+
+.. index:: function;call
+.. _calling_suffix_functions:
+
+Calling Suffix Functions
+------------------------
+
+There are two ways to call a suffix function:
+
+#. Suffix call.
+#. Function call.
+
+.. index:: ! literal suffix; suffix call syntax
+
+Suffix Call Syntax
+^^^^^^^^^^^^^^^^^^
+
+A suffix call has the same syntax as a literal with a :ref:`denomination<denominations>`:
+
+.. code-block:: solidity
+
+    42 suffix;
+    1.23 suffix;
+    "abc" suffix;
+    hex"12ff"suffix;
+    true suffix;
+
+The literal passed as input to the suffix function must be immediately followed by the name of the suffix.
+The two must be separated by whitespace unless it's a string, unicode or hexadecimal string literal,
+in which case the whitespace is optional.
+
+This call syntax supports only a single literal argument.
+Variables or expressions (even as simple as wrapping the literal in parentheses) are not allowed.
+Suffix functions defined with two parameters are also invoked with one literal - the
+:ref:`decomposition<fractional_decomposition>` of the literal into two values is performed implicitly
+by the compiler.
+
+.. note::
+    There are no negative number literals in Solidity.
+    A literal with a minus sign is an expression.
+    ``-123 suffix`` is equivalent to ``-(123 suffix)``, so ``suffix`` does not receive ``-123`` as input.
```

Comment:
I could imagine that some users will have a hard time while finding this out. not sure whether this note can be placed more prominent, so that a user may see this faster.

Reply:
we support info things in the error reporter right? do we actually use that right now for the normal compilation anywhere? I think it's mainly used in the the smt context, right? not sure whether any discussion was done on that, but I could imagine that this would be a nice info use-case. to just help a user to understand this faster. Not sure whether it makes a lot of sense, however, I leave that comment here.

Reply:
Sure! I was just wondering whether users got probably annoyed somehow to see that warning "always", where maybe info stuff could be just disabled - but I'm not sure whether we support that already

Reply:
I'll make it prominent.

> but I could imagine that this would be a nice info use-case. to just help a user to understand this faster. 

I think it would be too annoying to get an info message each time you try to use a negative literal. Especially that when you use a suffix from a library, it's you, not the library author who will be getting the warning :)